stream-chat-android-client/src/main/java/io/getstream/chat/android/client/ChatClient.kt

```diff
@@ -496,11 +510,13 @@ public class ChatClient internal constructor(
     }
 
     public fun reconnectSocket() {
-        when (val state = clientStateService.state) {
-            is ClientState.Anonymous -> socket.connectAnonymously()
-            is ClientState.User -> socket.connect(state.user)
-            is ClientState.Idle -> {
+        when (socketStateService.state) {
+            is SocketState.Disconnected -> when (val userState = userStateService.state) {
+                is UserState.UserSet -> socket.connect(userState.user)
+                is UserState.Anonymous.AnonymousUserSet -> socket.connectAnonymously()
+                else -> error("Weird user state ${userState.javaClass.simpleName} without user being set!")
```

Comment:
Is this intentional exception throwing? If yes, maybe we could add `@Throws` annotation, but maybe it would be better to log it instead

Reply:
We did something similar with state management in ClientStateService previously. Such approach is tough but helps to prevent bugs and weird scenarios. I don't mind just logging. @JcMinarro what do you think?

Reply:
Yes, it shouldn't be possible. But compilator let it happens

Reply:
```suggestion
                else -> error("Invalid user state $userState without user being set!")
```

I'd like to adjust the wording here a bit, "weird user state" is not something I want our clients to see in their logs 😄  

Also no need for the `javaClass.simpleName` solution if we make all the state objects used here either objects or data classes, so that they have a nice `toString` representation.